### ISSUE TITLE
OnePlus: If bindservice returns false or the service connection call back is not called, the callback for AT or ATS is not called

### ISSUE THREAD

**Issue by addev-ashish** [AzureAD/azure-activedirectory-library-for-android#899]
Please ensure the following information is provided when logging a new issue:

<ul>
  <li>Expected behavior: Everytime AT or ATS is called, the success or error callback should be called</li>
  <li>Actual behavior: if https://github.com/AzureAD/azure-activedirectory-library-for-android/blob/35c3eb19c257b82ccd29fb256dc7d1f8e337be67/adal/src/main/java/com/microsoft/aad/adal/BrokerAccountServiceHandler.java#L318 returns false, then the callback is not called</li>
  <li>Steps to reproduce the problem: Its repro'ing in OnePlus3 with Android 7.1.1</li>
  <li>Specifications like the version of the library, operating system etc: ADAL 1.12.0, OnePlus3 7.1.1</li>
  <li>Relevant logs and traces: Waiting on logs from customer</li>
</ul>

**Comment by wuqi713** [comment]
I have this question also.
1. company portal  5.0.3743.0
2. ADAL 1.12.0
3. Android 6.0

**Comment by nazukj** [comment]
Blocked on device logs.

**Comment by trevorwang** [comment]
```
08-11 15:46:25.904 30920-30920/com.example.abc I/hwaps: JNI_OnLoad
08-11 15:46:25.984 30920-30920/com.example.abc V/AudioManager: playSoundEffect   effectType: 0
08-11 15:46:25.984 30920-30920/com.example.abc V/AudioManager: querySoundEffectsEnabled...
08-11 15:46:26.016 30920-30920/com.example.abc V/BrokerProxy: 2017-08-11 07:46:26-null-Misused of the checking function. This is function is only used to checking the broker permissions of API 22 or less. ver:1.13.0
08-11 15:46:26.026 30920-30920/com.example.abc V/AcquireTokenRequest: 2017-08-11 07:46:26-987c7e3f-8344-4e16-8abd-eae77ff00ddf-Sending async task from thread:30920 ver:1.13.0
08-11 15:46:26.029 30920-31016/com.example.abc V/AcquireTokenRequest: 2017-08-11 07:46:26-987c7e3f-8344-4e16-8abd-eae77ff00ddf-Running task in thread:31016 ver:1.13.0
08-11 15:46:26.030 30920-31016/com.example.abc V/AcquireTokenRequest: 2017-08-11 07:46:26-987c7e3f-8344-4e16-8abd-eae77ff00ddf-Start validating authority ver:1.13.0
08-11 15:46:26.030 30920-31016/com.example.abc V/AcquireTokenRequest: 2017-08-11 07:46:26-987c7e3f-8344-4e16-8abd-eae77ff00ddf-The passed in authority is valid. ver:1.13.0
08-11 15:46:26.042 30920-31016/com.example.abc V/AuthenticationContext: 2017-08-11 07:46:26-987c7e3f-8344-4e16-8abd-eae77ff00ddf-Broker redirectUri:msauth://com.example.abc/xxxxxxxxxxxxxxxxxD packagename:com.example.abc signatureDigest:xxxxxxxxxx= ver:1.13.0
08-11 15:46:26.044 30920-31016/com.example.abc V/AcquireTokenRequest:verifyBrokerRedirectUri: 2017-08-11 07:46:26-987c7e3f-8344-4e16-8abd-eae77ff00ddf-The broker redirect URI is valid: msauth://com.example.abc/xxxxxxxxxxxxxxxxx ver:1.13.0
08-11 15:46:26.045 30920-31016/com.example.abc V/AcquireTokenRequest: 2017-08-11 07:46:26-987c7e3f-8344-4e16-8abd-eae77ff00ddf-Try to acquire token silently, return valid AT or use RT in the cache. ver:1.13.0
08-11 15:46:26.045 30920-31016/com.example.abc V/AcquireTokenRequest: 2017-08-11 07:46:26-987c7e3f-8344-4e16-8abd-eae77ff00ddf-Try to silently get token from local cache. ver:1.13.0
08-11 15:46:26.046 30920-31016/com.example.abc V/TokenCacheAccessor: 2017-08-11 07:46:26-987c7e3f-8344-4e16-8abd-eae77ff00ddf-No access token exists. ver:1.13.0
08-11 15:46:26.047 30920-31016/com.example.abc V/AcquireTokenSilentHandler: 2017-08-11 07:46:26-987c7e3f-8344-4e16-8abd-eae77ff00ddf-No valid access token exists, try with refresh token. ver:1.13.0
08-11 15:46:26.047 30920-31016/com.example.abc V/AcquireTokenSilentHandler: 2017-08-11 07:46:26-987c7e3f-8344-4e16-8abd-eae77ff00ddf-Regular token cache entry does not exist, try with MRRT. ver:1.13.0
08-11 15:46:26.048 30920-31016/com.example.abc V/AcquireTokenSilentHandler: 2017-08-11 07:46:26-987c7e3f-8344-4e16-8abd-eae77ff00ddf-MRRT token does not exist, try with FRT ver:1.13.0
08-11 15:46:26.049 30920-31016/com.example.abc V/AcquireTokenSilentHandler: 2017-08-11 07:46:26-987c7e3f-8344-4e16-8abd-eae77ff00ddf-FRT cache item does not exist, fall back to try MRRT. ver:1.13.0
08-11 15:46:26.049 30920-31016/com.example.abc V/AcquireTokenSilentHandler: 2017-08-11 07:46:26-987c7e3f-8344-4e16-8abd-eae77ff00ddf-Send request to use MRRT for new AT. ver:1.13.0
08-11 15:46:26.050 30920-31016/com.example.abc V/AcquireTokenSilentHandler: 2017-08-11 07:46:26-987c7e3f-8344-4e16-8abd-eae77ff00ddf-MRRT does not exist, cannot proceed with MRRT for new AT. ver:1.13.0
08-11 15:46:26.056 30920-31016/com.example.abc D/AcquireTokenRequest: Cannot get AT from local cache, switch to Broker for auth, clear tokens from local cache for the user.
08-11 15:46:26.058 30920-31016/com.example.abc V/AcquireTokenRequest: 2017-08-11 07:46:26-987c7e3f-8344-4e16-8abd-eae77ff00ddf-No token items need to be deleted for the user. ver:1.13.0
08-11 15:46:26.064 30920-31016/com.example.abc I/AcquireTokenWithBrokerRequest: 2017-08-11 07:46:26-987c7e3f-8344-4e16-8abd-eae77ff00ddf-Broker app is: com.microsoft.windowsintune.companyportal;Broker app version: VersionName=5.0.3772.0;VersonCode=1906233. ver:1.13.0 
08-11 15:46:26.064 30920-31016/com.example.abc V/AcquireTokenWithBrokerRequest: 2017-08-11 07:46:26-987c7e3f-8344-4e16-8abd-eae77ff00ddf-User is not specified, skipping background(silent) token request ver:1.13.0
08-11 15:46:26.065 30920-31016/com.example.abc D/AcquireTokenRequest: Trying to acquire token interactively.
08-11 15:46:26.075 30920-31016/com.example.abc V/AuthenticationContext: 2017-08-11 07:46:26-987c7e3f-8344-4e16-8abd-eae77ff00ddf-Put waiting request: 265627776 CorrelationId: 987c7e3f-8344-4e16-8abd-eae77ff00ddf ver:1.13.0
08-11 15:46:26.087 30920-31016/com.example.abc V/AcquireTokenRequest: 2017-08-11 07:46:26-987c7e3f-8344-4e16-8abd-eae77ff00ddf-Launch activity for interactive authentication via broker with callback: 265627776 ver:1.13.0
08-11 15:46:26.088 30920-31016/com.example.abc V/AcquireTokenWithBrokerRequest: 2017-08-11 07:46:26-987c7e3f-8344-4e16-8abd-eae77ff00ddf-Launch activity for interactive authentication via broker. ver:1.13.0
08-11 15:46:26.093 30920-31016/com.example.abc I/AcquireTokenWithBrokerRequest: 2017-08-11 07:46:26-987c7e3f-8344-4e16-8abd-eae77ff00ddf-Broker app is: com.microsoft.windowsintune.companyportal;Broker app version: VersionName=5.0.3772.0;VersonCode=1906233. ver:1.13.0 
08-11 15:46:26.099 30920-31016/com.example.abc V/BrokerAccountServiceHandler: 2017-08-11 07:46:26-987c7e3f-8344-4e16-8abd-eae77ff00ddf-Binding to BrokerAccountService for caller uid: 10272 ver:1.13.0
```

**Comment by trevorwang** [comment]
@nazukj I have added the logs for 20 days, could you please take a look on this?

**Comment by piqiums** [comment]
We tried to repro this issue on the reported devices but in vain. Now we made a fix to the case that leaks 'ServiceConnection' (reported by @trevorwang), and it will be released in V1.13.1. Here's the PR https://github.com/AzureAD/azure-activedirectory-library-for-android/pull/976 .  Any feedback on this fix will be appreciated. thanks

**Comment by trevorwang** [comment]
Hi @piqiums, it seems that 1.13.1 version doesn't contain this fix.

Due to this issue, the Company Portal can not be opened to  let user select account.
From your fix, I just saw an error posted to user. 
**_The user on this device cannot login anymore._**
